File: examples/sync/hyperliquid_public.py
import dcex


def main() -> None:
    client = dcex.hyperliquid()

    meta = client.get_meta()
    print(meta)

    orderbook = client.get_l2book(product_symbol="BTC-USD-SWAP")
    print(orderbook)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/hyperliquid_public.rs
use std::time::Duration;

use dcex::exchanges::hyperliquid::HyperliquidClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = HyperliquidClient::public(false, Duration::from_secs(10))?;

    let meta = client.get_meta().await?;
    println!("{}", meta.data);

    let orderbook = client
        .get_l2book_with(vec![(
            "product_symbol".to_string(),
            "BTC-USD-SWAP".to_string(),
        )])
        .await?;
    println!("{}", orderbook.data);

    Ok(())
}
